sympy/series/tests/test_formal.py

```diff
@@ -531,23 +542,42 @@ def test_fps__composition():
     raises(ValueError, lambda: f1.compose(f3, x))
     raises(ValueError, lambda: f2.compose(f3, x))
 
-    assert f1.compose(f2, x) == 1 + x + x**2/2 - x**4/8 - x**5/15 + O(x**6)
-    assert f1.compose(f2, x, n=4) == 1 + x + x**2/2 + O(x**4)
-    assert f1.compose(f2, x, n=8) == \
+    fcomp = f1.compose(f2, x)
+    assert isinstance(fcomp, FormalPowerSeriesCompose)
+    assert fcomp.outer_func == exp(x)
+    assert fcomp.inner_func == sin(x)
+    assert fcomp._eval_terms(6) == 1 + x + x**2/2 - x**4/8 - x**5/15
+    assert fcomp.truncate() == 1 + x + x**2/2 - x**4/8 - x**5/15 + O(x**6)
+    assert fcomp.truncate(5) == 1 + x + x**2/2 - x**4/8 + O(x**5)
+
+    raises(NotImplementedError, lambda: fcomp._eval_term(5))
+
+    assert f1.compose(f2, x).truncate(4) == 1 + x + x**2/2 + O(x**4)
+    assert f1.compose(f2, x).truncate(8) == \
         1 + x + x**2/2 - x**4/8 - x**5/15 - x**6/240 + x**7/90 + O(x**8)
+    assert f1.compose(f2, x).truncate(6) == \
+        1 + x + x**2/2 - x**4/8 - x**5/15 + O(x**6)
+
+    assert f2.compose(f2, x).truncate(4) == x - x**3/3 + O(x**4)
+    assert f2.compose(f2, x).truncate(8) == x - x**3/3 + x**5/10 - 8*x**7/315 + O(x**8)
+    assert f2.compose(f2, x).truncate(6) == x - x**3/3 + x**5/10 + O(x**6)
 
-    assert f2.compose(f2, x, n=4) == x - x**3/3 + O(x**4)
-    assert f2.compose(f2, x, n=8) == x - x**3/3 + x**5/10 - 8*x**7/315 + O(x**8)
 
 def test_fps__inverse():
     f1, f2, f3 = fps(sin(x)), fps(exp(x)), fps(cos(x))
 
     raises(ValueError, lambda: f1.inverse(x))
-    raises(ValueError, lambda: f1.inverse(x, n=8))
 
-    assert f2.inverse(x) == 1 - x + x**2/2 - x**3/6 + x**4/24 - x**5/120 + O(x**6)
-    assert f2.inverse(x, n=8) == \
+    finv = f2.inverse(x)
+    assert isinstance(finv, FormalPowerSeriesInverse)
+    assert finv.function == exp(-x)
+    assert finv._eval_terms(5) == 1 - x + x**2/2 - x**3/6 + x**4/24
+    assert finv.truncate() == 1 - x + x**2/2 - x**3/6 + x**4/24 - x**5/120 + O(x**6)
+    assert finv.truncate(5) == 1 - x + x**2/2 - x**3/6 + x**4/24 + O(x**5)
+
+    raises(NotImplementedError, lambda: finv._eval_term(5))
+    assert f2.inverse(x).truncate(8) == \
         1 - x + x**2/2 - x**3/6 + x**4/24 - x**5/120 + x**6/720 - x**7/5040 + O(x**8)
 
-    assert f3.inverse(x) == 1 + x**2/2 + 5*x**4/24 + O(x**6)
-    assert f3.inverse(x, n=8) == 1 + x**2/2 + 5*x**4/24 + 61*x**6/720 + O(x**8)
+    assert f3.inverse(x).truncate() == 1 + x**2/2 + 5*x**4/24 + O(x**6)
+    assert f3.inverse(x).truncate(8) == 1 + x**2/2 + 5*x**4/24 + 61*x**6/720 + O(x**8)
```

Comment:
Also add tests for anything that should is not supported or not implemented yet.